src/app/home/components/StatusContainer/hook/useStatusContainer/useStatusContainer.ts

```diff
@@ -0,0 +1,97 @@
+import { useCallback, useEffect, useState } from "react";
+
+import fetchNearbyStation from "@/app/home/api/fetchNearbyStation";
+import { convertLatLonToTM, getDetailedAddress, showErrorMessage } from "@/app/home/utils";
+import { WEATHER_KEY } from "@/lib/api/queryKeys";
+import {
+  useLocationStore,
+  useSaveTimeStore,
+  useWeatherStore,
+} from "@/lib/stores/useLocationDataStore";
+import useUserLocationStore from "@/lib/stores/useUserLocationStore";
+import { useQuery } from "@tanstack/react-query";
+
+import locationCompare from "../../utils/locationCompare";
+import timeCompare from "../../utils/timeCompare";
+
+const useStatusContainer = () => {
+  const [isClient, setIsClient] = useState(false);
+  const [isFetchRequired, setIsFetchRequired] = useState(false);
+  const { location, setLocation } = useLocationStore();
+  const { weatherData, setWeatherData } = useWeatherStore();
+  const { time, setTime } = useSaveTimeStore();
+
+  const { userAddress, setUserLocation, setUserAddress } = useUserLocationStore();
+
+  const { data: weatherInfo, isLoading } = useQuery({
+    queryKey: [WEATHER_KEY.GET_WEATHER_DATA],
+    queryFn: () => fetchNearbyStation(location),
+    enabled:
+      !!location.lat && !!location.lng && !!location.tmX && !!location.tmY && isFetchRequired,
+  });
+
+  const showYourLocation = ({ coords }: GeolocationPosition) => {
+    const lat = coords.latitude;
+    const lng = coords.longitude;
+    const { x: tmX, y: tmY } = convertLatLonToTM(lat, lng);
+
+    const isLocationCompare = locationCompare(location, { lat, lng, tmX, tmY });
+    const isTimeCompare = timeCompare(time, 30);
+
+    if (isLocationCompare || isTimeCompare) {
+      setIsFetchRequired(true);
+      setTime(new Date());
+      setLocation({ lat, lng, tmX, tmY });
+      setUserLocation({ lat, lng });
+    }
+  };
+
+  useEffect(() => {
+    if (!isClient) {
+      return;
+    }
+
+    navigator.geolocation.getCurrentPosition(showYourLocation, showErrorMessage);
+  }, [isClient]);
+
+  useEffect(() => {
+    if (!isClient) {
+      return;
+    }
+
+    if (weatherInfo && JSON.stringify(weatherData) !== JSON.stringify(weatherInfo)) {
+      setWeatherData(weatherInfo);
+    }
+  }, [weatherInfo, isClient]);
+
+  useEffect(() => {
+    setIsClient(true);
+  }, []);
+
+  const GetDetailedAddress = useCallback(async () => {
+    if (!location.lat || !location.lng) {
+      return;
+    }
+
+    const MyAddress = await getDetailedAddress({ lat: location.lat, lng: location.lng });
+
+    if (!MyAddress) {
+      return;
+    }
+
+    setUserAddress(MyAddress);
+  }, [location.lat, location.lng]);
+
+  useEffect(() => {
+    GetDetailedAddress();
+  }, [GetDetailedAddress]);

```

Comment:
[ ASK ] 이 부분은 무조건 API 호출이 한번은 꼭 일어나야하는건가요?

만약 다른곳에서도 getDetailedAddress 를 통해 userAddress를 호출해주는곳이 있다면

```suggestion
  useEffect(() => {
    if(userAddress){
      return
    }
    GetDetailedAddress();
  }, [GetDetailedAddress]);

```
API 호출을 방지해줘도 될것같긴한데 해당 Store의 사용처를 잘몰라서 궁금하네요!

Reply:
생각해 보니까 `GetDetailAddress` 함수 호출도 `showYourLocation` 내부에서 함께 동작하도록 호출 수를 개선할 수 있을 것 같아요 ! 👍